python/tidy/servo_tidy/tidy.py

```diff
@@ -631,28 +631,55 @@ def check_webidl_spec(file_name, contents):
     yield (0, "No specification link found.")
 
 
+class DuplicateKeyError(StandardError):
+    def __init__(self, key):
+        self.key = key
+
+
+class UnsortedKeyError(StandardError):
+    def __init__(self, a, b):
+        self.a = a
+        self.b = b
+
+
 def check_for_possible_duplicate_json_keys(key_value_pairs):
     keys = [x[0] for x in key_value_pairs]
     seen_keys = set()
     for key in keys:
         if key in seen_keys:
-            raise KeyError(key)
+            raise DuplicateKeyError(key)
 
         seen_keys.add(key)
 
 
+def check_for_alphabetical_sorted_json_keys(key_value_pairs):
+    for a, b in itertools.combinations(key_value_pairs, 2):
+        if a[0] > b[0]:
+            raise UnsortedKeyError(a[0], b[0])
+
+
+def check_json_requirements(filename):
+    def check_fn(key_value_pairs):
+        check_for_possible_duplicate_json_keys(key_value_pairs)
+        if filename.startswith("./resources/"):
```

Comment:
Having something like this would be specific to Servo, which we don't want, because tidy is published every once in a while, and is being used by other projects. We have a config file in the root for this (`servo-tidy.toml`). But, I doubt this would be necessary. We can check for sorting and duplicated keys at the same time.